lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -1870,3 +1889,49 @@ invariant msg a predicate =
 distance :: (Ord a, Num a) => a -> a -> a
 distance a b =
     if a < b then b - a else a - b
+
+data PoolMetadataSource
+    = None
+    | FetchDirect
+    | FetchSMASH URI
+    deriving (Show, Generic, Eq)
+
+instance ToText PoolMetadataSource where
+    toText None = (T.pack "")
+    toText FetchDirect = (T.pack "direct")
+    toText (FetchSMASH uri) = T.pack $ uriToString id uri ""
+
+instance FromText PoolMetadataSource where
+    fromText "" = Right None
+    fromText "direct" = Right FetchDirect
+    fromText (T.unpack -> uri) =
+        case parseURI uri of
+            Nothing -> Left $ TextDecodingError ("Could not parse URI: " <> uri)
+            Just uri' -> Right (FetchSMASH uri')
+
+data Settings = Settings {
+    poolMetadataSource :: PoolMetadataSource
+} deriving (Show, Generic, Eq)
+
+defaultSettings :: Settings
+defaultSettings = Settings {
+    poolMetadataSource = None
+}
+
+instance FromJSON PoolMetadataSource where
```

Comment:
What we usually do is define aeson instances on `ApiT PoolMetadataSource` and
```haskell
data ApiSettings = ApiSettings { poolMetadataSource :: !(ApiT PoolMetadataSource) }
```
That way affords different JSON encodings for primitive types, if needed.

Reply:
There is an instance for `ApiT Settings`:

```hs
instance FromJSON (ApiT W.Settings) where
    parseJSON = fmap ApiT . genericParseJSON defaultRecordTypeOptions
instance ToJSON (ApiT W.Settings) where
    toJSON = genericToJSON defaultRecordTypeOptions . getApiT
```

Reply:
That's rule #154 indeed, no JSON instances in `Primitive.Types`. JSON instances for the API interfaces are located only in API.Types.